Dockerfile

```diff
@@ -252,16 +253,17 @@ WORKDIR /go/src/github.com/containerd/containerd
 ARG DEBIAN_FRONTEND
 ARG TARGETPLATFORM
 RUN xx-apt-get install -y --no-install-recommends gcc libbtrfs-dev libsecret-1-dev
+ARG DOCKER_STATIC
 RUN --mount=from=containerd-src,src=/usr/src/containerd,rw \
     --mount=type=cache,target=/root/.cache <<EOT
   set -e
   export CC=$(xx-info)-gcc
-  export CGO_ENABLED=0
+  export CGO_ENABLED=$([ "$DOCKER_STATIC" = "1" ] && echo "0" || echo "1")
```

Comment:
Yes needed otherwise ctr is not static and containerd build warns for native builds:

```
$ docker buildx build --target containerd .
...
#22 [containerd-build 3/3] RUN --mount=from=containerd-src,src=/usr/src/containerd,rw     --mount=type=cache,target=/root/.cache <<EOT (set -e...)
#22 0.862 + bin/ctr
#22 1.921 + bin/containerd
#22 3.631 # github.com/containerd/containerd/cmd/containerd
#22 3.631 /usr/bin/ld: /tmp/go-link-508382094/000019.o: in function `New':
#22 3.631 /go/src/github.com/containerd/containerd/vendor/github.com/miekg/pkcs11/pkcs11.go:77: warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
#22 3.712 + bin/containerd-stress
#22 4.438 + bin/containerd-shim
#22 4.885 + bin/containerd-shim-runc-v1
#22 5.423 + bin/containerd-shim-runc-v2
#22 5.767 + binaries
#22 DONE 5.9s
```

Reply:
Looks like something we should fix in containerd; they added the `STATIC=1` make var; I think that one should ideally take care of that automatically.